test/functional/feature_bind_port_externalip.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test that the proper port is used for -externalip=
+"""
+
+from test_framework.test_framework import BitcoinTestFramework, SkipTest
+from test_framework.util import assert_equal
+
+# We need to bind to a routable address for this test to exercise the relevant code.
+# To set a routable IP address on the machine use:
+# Linux:
+# ifconfig lo:0 1.1.1.1/32 up  # to set up
+# ifconfig lo:0 down  # to remove it, after the test
+# FreeBSD:
+# ifconfig lo0 1.1.1.1/32 alias  # to set up
+# ifconfig lo0 1.1.1.1 -alias  # to remove it, after the test
+addr = '1.1.1.1'
+
+# array of tuples [arguments, expected port in localaddresses]
+expected = [
+    [['-externalip=2.2.2.2',       '-port=20001'                               ], 20001],
+    [['-externalip=2.2.2.2',       '-port=20002', '-bind={}'.format(addr)],       20002],
+    [['-externalip=2.2.2.2',                      '-bind={}'.format(addr)],       11002],
+    [['-externalip=2.2.2.2',       '-port=20003', '-bind={}:20004'.format(addr)], 20004],
+    [['-externalip=2.2.2.2',                      '-bind={}:20005'.format(addr)], 20005],
+    [['-externalip=2.2.2.2:20006', '-port=20007'                               ], 20006],
+    [['-externalip=2.2.2.2:20008', '-port=20009', '-bind={}'.format(addr)],       20008],
+    [['-externalip=2.2.2.2:20010',                '-bind={}'.format(addr)],       20010],
+    [['-externalip=2.2.2.2:20011', '-port=20012', '-bind={}:20013'.format(addr)], 20011],
+    [['-externalip=2.2.2.2:20014',                '-bind={}:20015'.format(addr)], 20014],
+    [['-externalip=2.2.2.2',       '-port=20016', '-bind={}:20017'.format(addr),
+                                             '-whitebind={}:20018'.format(addr)], 20017],
+    [['-externalip=2.2.2.2',       '-port=20019',
+                                             '-whitebind={}:20020'.format(addr)], 20020],
+]
+
+class BindPortTest(BitcoinTestFramework):
+    def set_test_params(self):
+        # Avoid any -bind= on the command line. Force the framework to avoid adding -bind=127.0.0.1.
+        self.setup_clean_chain = True
+        self.bind_to_localhost_only = False
+        self.num_nodes = len(expected)
+
+    def add_options(self, parser):
+        parser.add_argument(
+            "--ihave1111", action='store_true', dest="ihave1111",
+            help="Run the test, assuming {} is configured on the machine".format(addr),
+            default=False)
+
+    def setup_nodes(self):
```

Comment:
Does this need to be overwritten? You can use `skip_test_if_missing_module`

Reply:
`raise SkipTest()` is not the main reason to override `setup_nodes()`. It is necessary to specify the command line arguments for each bitcoind node. The current test does that by `self.add_nodes(..., extra_args=my_variable_here)` in `setup_nodes()`. Other tests do that either in `setup_nodes()` or `setup_network()`.

Is there a better way?

Reply:
Set `self.extra_args = ...`?

Reply:
Indeed! Changed, thanks for the suggestion!